neo/Trie/MPT/MPTReadOnlyTrie.cs

```diff
@@ -0,0 +1,144 @@
+using Neo.IO;
+using System;
+using System.Collections.Generic;
+
+namespace Neo.Trie.MPT
+{
+    public class MPTReadOnlyTrie
+    {
+        private readonly MPTReadOnlyDb rodb;
+        protected MPTNode root;
+
+        public MPTReadOnlyTrie(UInt256 root, IKVReadOnlyStore store)
+        {
+            if (store is null)
+                throw new System.ArgumentNullException();
+
+            this.rodb = new MPTReadOnlyDb(store);
+            if (root is null)
+            {
+                this.root = HashNode.EmptyNode();
+            }
+            else
+            {
+                this.root = new HashNode(root);
+            }
+        }
+
+        public MPTNode Resolve(HashNode hn)
+        {
+            return rodb.Node(hn.Hash);
+        }
+
+        public bool TryGet(byte[] key, out byte[] value)
+        {
+            var path = key.ToNibbles();
+            return TryGet(ref root, path, out value);
+        }
+
+        private bool TryGet(ref MPTNode node, byte[] path, out byte[] value)
+        {
+            switch (node)
+            {
+                case LeafNode leafNode:
+                    {
+                        if (path.Length == 0)
+                        {
+                            value = (byte[])leafNode.Value.Clone();
+                            return true;
+                        }
+                        break;
+                    }
+                case HashNode hashNode:
+                    {
+                        if (hashNode.IsEmptyNode) break;
+                        var newNode = Resolve(hashNode);
+                        if (newNode is null) break;
+                        node = newNode;
+                        return TryGet(ref node, path, out value);
+                    }
+                case BranchNode branchNode:
+                    {
+                        if (path.Length == 0)
+                        {
+                            return TryGet(ref branchNode.Children[16], path, out value);
+                        }
+                        return TryGet(ref branchNode.Children[path[0]], path.Skip(1), out value);
```

Comment:
Could you add comments about why this skip and offsets?

Reply:
![canvas](https://user-images.githubusercontent.com/22324403/79317428-4920bd80-7f38-11ea-9660-39203c357570.png)
Take this graph for example.
If the path is `0a0c` you get `val` and the value is `v1`
If the path is longer than `0a0c` like `0a0c0?*`,  `?` is from 0 to f, we find the value from the corresponding branch.